docs/developer/architecture/user_customisations.rst

```diff
@@ -0,0 +1,51 @@
+User Customisations and Persistence
+===================================
+
+Context
+-------
+
+MetalK8s comes with a set of tools and services that may need to be configured
+on site. At the same time, we don't want the administrator of the cluster to
+master each and every service of the cluster. We also don't want to allow any
+kind of customisations since it would make the system even more complex to test
+and maintain over the time.
+
+In addition to those services, MetalK8s deployment may have to be adapted
+depending on the architecture of the platform or depending on the different
+use cases running on the top of it.
+
+Be it services or MetalK8s customisations, we need to ensure it is persisted and
+resistant to various type of events such as node reboot, upgrade, downgrade,
+backup, restore
+
+.. _usercusto-requirements:
+
+Requirements
+------------
+
+As a MetalK8s administrator, I have access to a finite list of settings I can
+customise on-site in order to match with my environment specificities:
+
+- list of static users and credentials configured in Dex
+- Integration with an external IDP configuration in Dex
+- Alert notification configuration in Alert Manager
+- Number of replicas for the Prometheus and Alert Manager cluster
+- Other items will appear as we add new customisable features in MetalK8s
+
+Depending on the the setting, impact of the customisation can be at cluster
+level or namespace level (not sure it is needed)
+
+As a MetalK8s administrator, I have a uniq CLI/API way of doing MetalK8s
```

Comment:
Can we specify what "way of doing" means? Is it a single entrypoint (`kubectl edit cm ...`, or `kubectl metalk8s userconfig ...` if we have our CLI) ? A single customisation format? Should it be even a single "source of config", like a single file?

Reply:
So I was mainly thinking of common way of updating one setting using `kubectl edit cm` indeed. will update. Does not mean it should be the only way but at least we always have this backing door